app/controllers/email_authentications/passwords_controller.rb

```diff
@@ -0,0 +1,7 @@
+class EmailAuthentications::PasswordsController < Devise::PasswordsController
+  def edit
+    users_email_authentication = resource_class.with_reset_password_token(params["reset_password_token"])
```

Comment:
It's tempting to make a pull request on devise for this change. I don't think it would be approved because it would probably cause trouble for existing apps, but it seems like the better flow to me.

Reply:
It's tempting indeed 😅 You could always introduce a configurable option, and default it to the current behavior. That would leave existing apps untouched. I'm curious if folks have tried opening a PR for this before. It seems like a fairly common user flow - let the user know early that their attempt won't work.